app/services/image_search_service.py

```
@@ -0,0 +1,41 @@
+        """
+        # 키워드 추출
+        keywords_dict = VisionService.extract_keywords(contents)
+
+        # 키워드 정규화
+        keyword_list = VisionService.normalize_keywords(keywords_dict)
+
+        # 종목 추론
+        try:
+            brand_name = GPTService.infer_brand_name_from_keywords(keyword_list)
+        except Exception:
+            raise APIException(ErrorStatus.GPT_API_ERROR)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)